fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/federation/LinkedOrganizationsInviteFragment.kt

```diff
@@ -141,20 +142,17 @@ class LinkedOrganizationsInviteFragment : Fragment() {
   }
 
   private fun finishButton() {
-    val repo = linkedOrganizationsViewModel.getRepository()
-    if (repo.otherLaoId == null ||
-        repo.otherServerAddr == null ||
-        repo.otherPublicKey == null ||
-        repo.getChallenge() == null) {
+    if (!linkedOrganizationsViewModel.isRepositoryValid() ||
+        linkedOrganizationsViewModel.getChallenge() == null) {
       logAndShow(requireContext(), TAG, R.string.error_invalid_federation_info)
     } else {
       laoViewModel.addDisposable(
           linkedOrganizationsViewModel
               .sendFederationInit(
-                  repo.otherLaoId!!,
-                  repo.otherServerAddr!!,
-                  repo.otherPublicKey!!,
-                  repo.getChallenge()!!,
+                  linkedOrganizationsViewModel.getOtherLao()!!,
+                  linkedOrganizationsViewModel.getOtherServerAddress()!!,
+                  linkedOrganizationsViewModel.getOtherPublicKey()!!,
+                  linkedOrganizationsViewModel.getChallenge()!!,
               )
```

Comment:
This seems very odd, you're calling a method in the view model (sendFederationInit and passing arguments by calling again methods of the view model? Why isn't it done entirely in the view model?

Reply:
Yes doing it in the view model is actually way better, I can then also remove some getters that are only used for that